### PR TITLE
Allow deleting Rewind credentials from Settings/Security/Backups

### PR DESCRIPTION
**Pull request by rcoll**
- Adds delete button to CredentialsForm
- Adds data-layer logic for deleting credentials

Testing:
1. To test, you will need to sandbox the WPCOM api, then the VP API from WPCOM, with D8041 applied to VaultPress and D8042 applied to wpcom.
2. Navigate to site-settings/security/backups and input credentials into the form. Save the form.
3. You should now see a delete button when you expand the foldable card. Click it, and after a short period of time, your credentials should be deleted.

### CODE REVIEW

File: client/state/data-layer/wpcom/activity-log/delete-credentials/index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+};
+
+export const success = ( { dispatch }, action ) => {
+	dispatch( {
+		type: JETPACK_CREDENTIALS_STORE,
+		credentials: {
+			main: null,
+		},
+		siteId: action.siteId,
+	} );
```
Comment (dmsnell) [inline client/state/data-layer/wpcom/activity-log/delete-credentials/index.js]: If I were to guess what kind of action would feed into the reducers then I would expect a `_DELETE` action to eliminate the credentials from the state tree, not necessarily a `_STORE`.

In fact, we have the opportunity here in the action handler to just send out the HTTP call to delete the credentials while letting the reducers do the actual removal on this `_DELETE` action. If the network request fails then we can add the credentials back in with a `_STORE` generated from the original action.

At the time I don't see us actually removing the credentials with any network calls.

File: client/state/data-layer/wpcom/activity-log/delete-credentials/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	);
+};
+
+export const failure = ( { dispatch }, action ) => {
+	dispatch(
+		errorNotice( i18n.translate( 'Error deleting credentials. Please try again.' ), {
+			duration: 4000,
+			id: action.noticeId,
+		} )
+	);
```
Comment (dmsnell) [inline client/state/data-layer/wpcom/activity-log/delete-credentials/index.js]: See comment above: this is a nice message, but we can inject the originally-deleted credentials back into state by "undoing" the delete action -> `_STORE`

File: client/state/data-layer/wpcom/activity-log/delete-credentials/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { JETPACK_CREDENTIALS_DELETE, JETPACK_CREDENTIALS_STORE } from 'state/action-types';
+import { successNotice, errorNotice } from 'state/notices/actions';
+
+export const request = ( { dispatch }, action ) => {
+	const notice = successNotice( i18n.translate( 'Deleting credentials…' ), { duration: 4000 } );
```
Comment (dmsnell) [inline client/state/data-layer/wpcom/activity-log/delete-credentials/index.js]: Do we need to issue these notices on this? In general I haven't seen us follow this idiom around Calypso. Maybe a designer could provide some helpful advice here.

**Comment by dmsnell** [comment]
@rcoll the form code looks great but I have questions about the notices we're creating - not sure they are necessary. I added the design-review label in hopes we can get some additional feedback on that pattern.

was there something about deleting credentials that led you to think we needed the additional signaling to our customers?

File: client/my-sites/site-settings/jetpack-credentials/index.jsx [hunk 4]
```diff
@@ -26,7 +26,8 @@ import {
 } from 'state/selectors';
 import {
 	updateCredentials as updateCredentialsAction,
-	autoConfigCredentials as autoConfigCredentialsAction
+	autoConfigCredentials as autoConfigCredentialsAction,
+	deleteCredentials as deleteCredentialsAction
```
Comment (eliorivero) [inline client/my-sites/site-settings/jetpack-credentials/index.jsx]: This renaming was discouraged by @dmsnell and I agree. It's not too useful and only pollutes things.

**Comment by MichaelArestad** [comment]
@rcoll Let's do something more like this:

image

* This uses a clickable card
* When clicked, expands to this:

image

**Comment by MichaelArestad** [comment]
This should work for automatically connected sites:

image

File: client/state/data-layer/wpcom/activity-log/delete-credentials/index.js (hunk 1 above)
Comment (rcoll) [reply client/state/data-layer/wpcom/activity-log/delete-credentials/index.js]: I agree. Do you think this is a blocker for this now, or is it something we can think about optimizing when we retrofit with the state-machine design?

File: client/my-sites/site-settings/jetpack-credentials/credentials-configured/index.jsx [hunk 5]
```diff
@@ -40,15 +49,48 @@ class CredentialsConfigured extends Component {
...
+						>
+							<Gridicon
+								className="credentials-configured__revoke-icon"
+								icon="link-break"
+								size={ 18 }
+							/>
+							{ translate( 'Revoke credentials' ) }
+						</Button>
+						<Button primary onClick={ this.toggleRevoking }>
+							{ translate( 'Stay connected' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/jetpack-credentials/credentials-configured/index.jsx]: Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Stay Connected' )` *ES Score: 17*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**